From eaec6c59518da090ac48ace12ca93a178672a59f Mon Sep 17 00:00:00 2001
From: 0p1pp1 <0p1pp1@fms.freenet>
Date: Sun, 3 Apr 2016 19:12:25 +0900
Subject: [PATCH 2/5] mpegts: check current/next flag when parsing sections

Signed-off-by: 0p1pp1 <0p1pp1@fms.freenet>
---
 libavformat/mpegts.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libavformat/mpegts.c b/libavformat/mpegts.c
index 65824dd..1db9858 100644
--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -1852,6 +1854,8 @@ static void pmt_cb(MpegTSFilter *filter, const uint8_t *section, int section_len
         return;
     if (skip_identical(h, tssf))
         return;
+    if (!h->current)
+        return;
 
     av_log(ts->stream, AV_LOG_TRACE, "sid=0x%x sec_num=%d/%d version=%d\n",
             h->id, h->sec_num, h->last_sec_num, h->version);
@@ -2018,6 +2022,8 @@ static void pat_cb(MpegTSFilter *filter, const uint8_t *section, int section_len
         return;
     if (ts->skip_changes)
         return;
+    if (!h->current)
+        return;
 
     if (skip_identical(h, tssf))
         return;
@@ -2095,6 +2101,8 @@ static void sdt_cb(MpegTSFilter *filter, const uint8_t *section, int section_len
         return;
     if (skip_identical(h, tssf))
         return;
+    if (!h->current)
+        return;
 
     onid = get16(&p, p_end);
     if (onid < 0)
-- 
2.8.0

